packages/trader/src/Modules/Trading/Helpers/contract-type.tsx

```diff
@@ -5,7 +5,7 @@ import { TContractType, TContractCategory, TList } from '../Components/Form/Cont
 type TContractTypesList = {
     [key: string]: {
         name: string;
-        categories: TContractType[];
+        categories: DeepRequired<TContractType[]>;
```

Comment:
What is the usage of 'DeepRequired'?

Reply:
AFAIK, it makes all properties of an object required in-depth.
@henry-deriv could you please elaborate🙏🏻?

Reply:
DeepRequired is to make sure everything in the object is not optional. For example, lets say we have deeply nested object like below. `Required<ContractUpdate>` only makes `stop_loss` and `take_profit` not optional while its own property inside are optional. `DeepRequired<ContractUpdate>` makes the whole object including nested objects and its properties required.

```
**### Original Object**
ContractUpdate = {  
    stop_loss?: {
        display_name?: string;
        order_amount?: null | number;
        order_date?: number;
        value?: null | string;
    };
  take_profit?: {
    display_name?: string;
    order_amount?: null | number;
    order_date?: number;
    value?: null | string;
  };
} 
```

```
**### Object if you use `Required`**
ContractUpdate = {  
    stop_loss: {
        display_name?: string;
        order_amount?: null | number;
        order_date?: number;
        value?: null | string;
    };
  take_profit: {
    display_name?: string;
    order_amount?: null | number;
    order_date?: number;
    value?: null | string;
  };
} 
```

```
**### Object if you use `DeepRequired`**
ContractUpdate = {  
    stop_loss: {
        display_name: string;
        order_amount: null | number;
        order_date: number;
        value: null | string;
    };
  take_profit: {
    display_name: string;
    order_amount: null | number;
    order_date: number;
    value: null | string;
  };
} 
```